analyzers/src/SonarAnalyzer.Utilities/RuleDescriptors/RuleDetail.cs

```diff
@@ -76,20 +76,13 @@ public RuleDetail(AnalyzerLanguage language, ResourceManager resources, string i
 
         private static string HtmlDescription(AnalyzerLanguage language, string id)
         {
-            if (SonarAnalyzerUtilitiesAssembly.GetManifestResourceNames().FirstOrDefault(MatchesRule) is { } resourceName)
+            var suffix = language.LanguageName switch
             {
-                using var stream = SonarAnalyzerUtilitiesAssembly.GetManifestResourceStream(resourceName);
-                using var reader = new StreamReader(stream);
-                return reader.ReadToEnd();
-            }
-            else
-            {
-                throw new InvalidDataException($"Could not locate resource for rule {id}");
-            }
-
-            bool MatchesRule(string resource) =>
-                resource.EndsWith($"SonarAnalyzer.Rules.Description.{id}.html", StringComparison.OrdinalIgnoreCase)
-                || resource.EndsWith($"SonarAnalyzer.Rules.Description.{id}{language.ResourceSuffix}.html", StringComparison.OrdinalIgnoreCase);
+                LanguageNames.CSharp => $@"cs\{id}_c#.html",
+                LanguageNames.VisualBasic => $@"vbnet\{id}_vb.net.html",
+                _ => throw new UnexpectedLanguageException(language)
```

Comment:
This line is not coverable, because it already fails inside `.LangaugeName` (poor `AnalyzerLanguage` design)